src/Components/Publishing/Ads/AllowedUsers.ts

```diff
@@ -0,0 +1,29 @@
+import { get } from "lodash"
+import { data as sd } from "sharify"
+
+/**
+ * For now ads, served by Hashtag Labs are only to viewable in the following instances:
+ *  1) When the environment is NOT Production
+ *  2) When a user is an Admin
+ *  3) When a user has been added to the allowlist set via HASHTAG_LAB_ADS_ALLOWLIST
+ */
+
+export const getAllowedUsers = () => {
+  const allowedUsers = (sd.HASHTAG_LAB_ADS_ALLOWLIST || "")
+    .split(",")
+    .filter(Boolean)
+  const currentUser = get(sd, "CURRENT_USER.email", "")
+  const env = sd.HASHTAG_LAB_ADS_ENABLED
+  const isAdmin = get(sd, "CURRENT_USER.type", "") === "Admin" ? true : false
+
+  if (env === "production") {
+    return false
+  }
+  if (allowedUsers.includes(currentUser)) {
+    return true
+  }
+  if (isAdmin) {
+    return true
+  }
+  return false
```

Comment:
The return-types here, combined with the function name, is also a bit ambiguous language-wise. Thinking: similar to the comment above, it's common to encode the type of data into the name of something -- so

 ```js
const isEnabled = true // => true / false == boolean 
```
or
```js
const username = 'chris' // name => string
```
So thinking about the name of the function `getAllowedUsers`, it encodes a few assumptions:
1) `get` -> return something
1) `Allowed` -> there's some sort of filtering logic going on internally
1) `Users` -> a collection of things

Since it's only returning a single boolean value, the name should reflect that: something like `isUserAllowed` (it's common to prefix things that are booleans with "is".)

Reply:
Also, another possibility for evaluating a set of conditions is the array method [`some`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some). (Though the use of `if` conditions is totally fine and clear!):

```js
const isAllowed = [
  isAdmin,
  sd.HASHTAG_LAB_ADS_ENABLED,
  allowedUsers.includes(currentUser),
].some(condition => condition === true) // or .some(Boolean)

return isAllowed
```